controllers/apps/component/transformer_component_rbac.go

```diff
@@ -160,7 +162,7 @@ func createOrUpdateServiceAccount(transCtx *componentTransformContext,
 	synthesizedComp := transCtx.SynthesizeComponent
 
 	sa := factory.BuildServiceAccount(synthesizedComp, serviceAccountName)
-	if err := setCompOwnershipNFinalizer(transCtx.Component, sa); err != nil {
+	if err := intctrlutil.SetOwnership(transCtx.Component, sa, model.GetScheme(), ""); err != nil {
```

Comment:
If these sub-resources are destined to become orphans, why set ownership for them?

Reply:
Another component CR can adopt an orphaned rbac resource. So that if an object get updated (e.g. modified by a user), component controller can reconcile it back.

Reply:
That sounds tricky, maybe we should have a ComponentDefinitionController to manage the lifecycle of these RBAC resources, as @Y-Rookie mentioned.